addon/v1alpha1/types_addontemplate.go

```diff
@@ -0,0 +1,181 @@
+package v1alpha1
+
+import (
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	work "open-cluster-management.io/api/work/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope="Cluster"
+// +kubebuilder:printcolumn:name="ADDON NAME",type=string,JSONPath=`.spec.addonName`
+
+// AddOnTemplate is the Custom Resource object, it is used to describe
+// how to deploy the addon agent and how to register the addon.
+//
+// AddOnTemplate is a cluster-scoped resource, and will only be used
+// on the hub cluster.
+type AddOnTemplate struct {
```

Comment:
ManagedClusterAddOnTemplate?

Reply:
should it start with "addon" to be consistent with addonDeploymentConfig?

Reply:
hrm, that makes sense also